impeller/renderer/backend/metal/context_mtl.h

```diff
@@ -164,7 +164,9 @@ class ContextMTL final : public Context,
   std::shared_ptr<AllocatorMTL> resource_allocator_;
   std::shared_ptr<const Capabilities> device_capabilities_;
   std::shared_ptr<const fml::SyncSwitch> is_gpu_disabled_sync_switch_;
-  std::deque<PendingTasks> tasks_awaiting_gpu_;
+  std::mutex tasks_awaiting_gpu_mutex_;
```

Comment:
Please use `Mutex` and `Lock` from the impeller namespace in "impeller/base/thread.h". The standard library mutexes aren't guaranteed to have the capability decorations that `IPLR_GUARDED_BY` uses.

Reply:
done